From 7fb39cddd05d08bbe6eeb0d7beb93a5e1013acba Mon Sep 17 00:00:00 2001
From: Udayan Banerji <udayan.banerji@intel.com>
Date: Mon, 16 Dec 2013 19:00:43 -0800
Subject: Dalvik: Fix the Dalvik disassembler for const-wide bytecodes

BZ: 159347

The dvmCompilerFullDisassembler should print out vB_wide value when dumping
instructions of format kFmt51l.

Change-Id: Iaea412f2d37490d8743c8ff1a120cf2a3275af3a
Category: device enablement
Domain: AOSP.Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, need rework
Signed-off-by: Udayan Banerji <udayan.banerji@intel.com>
---
 vm/compiler/Dataflow.cpp |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/Dataflow.cpp b/vm/compiler/Dataflow.cpp
index 2c442f6..84d8a2a 100644
--- a/vm/compiler/Dataflow.cpp
+++ b/vm/compiler/Dataflow.cpp
@@ -1498,9 +1498,11 @@ char *dvmCompilerFullDisassembler(const CompilationUnit *cUnit,
                 case kFmt21s:        // op vAA, #+BBBB
                 case kFmt21h:        // op vAA, #+BBBB00000[00000000]
                 case kFmt31i:        // op vAA, #+BBBBBBBB
-                case kFmt51l:        // op vAA, #+BBBBBBBBBBBBBBBB
                     snprintf(buffer + offset, left, " #%#x", insn->vB);
                     break;
+                case kFmt51l:        // op vAA, #+BBBBBBBBBBBBBBBB
+                    snprintf(buffer + offset, left, " #%#llx", insn->vB_wide);
+                    break;
                 case kFmt21c:        // op vAA, thing@BBBB
                 case kFmt31c:        // op vAA, thing@BBBBBBBB
                     snprintf(buffer + offset, left, " @%#x", insn->vB);
-- 
1.7.4.1

